rdkPlugins/Networking/source/DnsmasqSetup.cpp

```diff
@@ -123,25 +187,46 @@ bool DnsmasqSetup::set(const std::shared_ptr<DobbyRdkPluginUtils> &utils,
 
     std::lock_guard<std::mutex> locker(mLock);
 
-    Netfilter::RuleSet ruleSet = constructRules(containerId);
-
     // install the iptables rules
-    if (!netfilter->appendRules(ruleSet))
+    if (helper->ipv4())
     {
-        AI_LOG_ERROR_EXIT("failed to setup netfilter rules for dns");
-        return false;
+        Netfilter::RuleSet ipv4RuleSet = constructRules(containerId, AF_INET);
+        if (!netfilter->appendRules(ipv4RuleSet, AF_INET))
+        {
+            AI_LOG_ERROR_EXIT("failed to setup netfilter rules for dns");
+            return false;
+        }
+    }
+    if (helper->ipv6())
+    {
+        Netfilter::RuleSet ipv6RuleSet = constructRules(containerId, AF_INET6);
+        if (!netfilter->appendRules(ipv6RuleSet, AF_INET6))
+        {
+            AI_LOG_ERROR_EXIT("failed to setup netfilter rules for dns");
+            return false;
+        }
     }
 
     // set the nameserver, if using a NAT network setup then set the bridge
     // device as the nameserver (iptables will re-direct to localhost)
     std::string content;
     if (networkType == NetworkType::Nat)
     {
-        content = "nameserver " BRIDGE_ADDRESS "\n";
+        content.append("nameserver " BRIDGE_ADDRESS "\n");
```

Comment:
It was left intentionally, but thinking about it now, it should not have been. It was useful in an edge case on my VM setup on a build server, but in reality, it should never be the case and at best, this would only lead to an unnecessary DNS query if configured to ipv6 only, so I'll add the `helper->ipv4()` check.